go/host/db/batches.go

```diff
@@ -33,6 +33,15 @@ func (db *DB) GetBatchHeader(hash gethcommon.Hash) (*common.Header, error) {
 
 // AddBatchHeader adds a batch's header to the known headers
 func (db *DB) AddBatchHeader(batch *common.ExtBatch) error {
+	// We check if the batch is already stored, to avoid incrementing the total transaction count twice for one batch.
+	_, err := db.GetBatchHeader(batch.Hash())
```

Comment:
We avoid storing a batch twice. This is good hygiene, and avoids us double-counting total transactions, below.